client/me/security-2fa-key/index.jsx

```diff
@@ -38,19 +38,31 @@ class Security2faKey extends Component {
 
 	addKeyStart = ( event ) => {
 		event.preventDefault();
-		this.setState( { addingKey: true } );
+		this.setState( { addingKey: true, errorMessage: null } );
 	};
 
 	addKeyRegister = () => {
 		this.getKeysFromServer();
 	};
 
 	deleteKeyRegister = ( keyData ) => {
-		wpcom.req.get(
-			'/me/two-step/security-key/delete',
-			{ credential_id: keyData.id },
-			this.getKeysFromServer
-		);
+		const { translate } = this.props;
+		this.setState( { errorMessage: false } );
+		wpcom.req.get( '/me/two-step/security-key/delete', { credential_id: keyData.id }, ( err ) => {
+			if ( null === err ) {
+				this.getKeysFromServer();
+			} else {
+				const errorMessage =
+					'invalid_operation' === err.error
+						? translate(
+								`Unable to delete the last WordPress.com security key while enhanced account security is active. Deleting it may result in losing access to your account. If you still want to remove it, please disable enhanced account security.`
+						  )
+						: translate( 'The key could not be deleted. Please try again later.' );
```

Comment:
Can they do something else too than "try later", and will trying later likely resolve it?

More actionable the error for customer, the better.

Reply:
Trying again later might fix things if the SQL query used to remove the key failed for some reason, but it might not on other edge cases.

Reply:
Seems fine then 👍